### PR TITLE
Upgrade Fuel ecosystem dependencies

### PR DESCRIPTION
**Pull request by lostman**
### Description

This PR updates Fuel dependencies.

### Testing steps

Please provide the _exact_ testing steps for the reviewer(s) if this PR requires testing.

### Changelog

* fuels-code-gen 0.43
* fuels-macros 0.43
* fuels-types 0.42
* Sway (for fuel-indexer-test) 0.40.1
* fuel-core 0.18.1
* fuel-core-client 0.18.1
* fuel-crypto 0.31.2
* fuels 0.43

### CODE REVIEW

**Review by ra0x3 — DISMISSED** [review]
🥇

File: Cargo.toml [hunk 1]
```diff
@@ -82,12 +82,13 @@ fuel-indexer-postgres = { version = "0.16.2", path = "./packages/fuel-indexer-da
 fuel-indexer-schema = { version = "0.16.2", path = "./packages/fuel-indexer-schema", default-features = false }
 fuel-indexer-types = { version = "0.16.2", path = "./packages/fuel-indexer-types" }
 fuel-indexer-utils = { version = "0.16.2", path = "./packages/fuel-indexer-utils" }
-fuel-tx = { version = "0.26", default-features = false }
-fuel-types = { version = "0.26", default-features = false }
-fuels = { version = "0.40", default-features = false }
+fuel-tx = { version = "0.31" }
```
Comment (ra0x3) [inline Cargo.toml:85]:
```suggestion
fuel-tx = { version = "0.31", default-features = false }
```

File: Cargo.toml [hunk 2]
```diff
@@ -82,12 +82,13 @@ fuel-indexer-postgres = { version = "0.16.2", path = "./packages/fuel-indexer-da
...
+fuel-types = { version = "0.31", features = ["serde"] }
```
Comment (ra0x3) [inline Cargo.toml:86]:
```suggestion
fuel-types = { version = "0.31", default-features = false, features = ["serde"] }
```

**Review by ra0x3 — CHANGES_REQUESTED** [review]
2 small comments related to not including default features

**Review by ra0x3 — APPROVED** [review]

**Review by deekerno — APPROVED** [review]
Very nice. 💯

**Pull request merged**